proxy/http/HttpSM.cc

```diff
@@ -116,6 +117,9 @@ milestone_update_api_time(TransactionMilestones &milestones, ink_hrtime &api_tim
     }
   }
 }
+
+// Unique state machine identifier
+std::atomic<int64_t> next_sm_id;
```

Comment:
As a note, `std::atomic` does no initialization - this is depending on file scope initialization to zero the counter.